### ISSUE TITLE
*Request* Reenable CustomModOption EVENTS_UNIT_CAPTURE

### ISSUE THREAD

**Issue by UndeadDevel** [LoneGazebo/Community-Patch-DLL#4993]
_1. Mod version (i.e Date - 4/23):_
11-30-18

_2. Request_
In trying to get whoward's mod "[Capture Great People](http://www.picknmixmods.com/mods/CivV/Global/Capture%20Great%20People.html)" to work and failing in that, I noticed that the issue stems from the CustomModOption EVENTS_UNIT_CAPTURE breaking unit capturing when enabled (it is disabled by default in VP). However, without this option, the unit capture event is not sent to the mod so that the mod cannot react to it.
Specifically, when enabling the option the mod will essentially work except that in my test case of capturing a worker the worker still belonged to the civ that produced it and just ran away to their city on the next turn; I got the notification, however (I modified some code such that it does this for all units) so I know the mod works with that option. I also tested with only the option enabled and nothing else (all other code commented out) and the unit capture failed, so it's definitely something in the DLL that leads to EVENTS_UNIT_CAPTURE breaking unit capturing.

So the request is this: is it possible to make unit capture work properly with this CustomModOption enabled? I just need the event to be sent so that the mod can process it.

Edit: I should probably add that when I asked whoward about this he said the mod worked fine with his custom DLL so it's likely changes to the DLL in VP that are at issue

**Comment by Padre19** [comment]
???
UPDATE CustomModOptions SET Value = 1 WHERE Name = 'EVENTS_UNIT_CAPTURE';

**Comment by LoneGazebo** [comment]
I think I've fixed, we'll see.

**Comment by JFDaly** [comment]
Seems to still be broken with the latest version (15th Jan), unfortunately. I was just about to report this issue anew :p

**Comment by LoneGazebo** [comment]
Which part isn’t working l?

**Comment by LoneGazebo** [comment]
I tested part of it and it worked.

**Comment by LoneGazebo** [comment]
Just looked, you have to fill out the <Capture> table in the Unit XML for a unit to have a capture type. I'm pretty sure it has always been that way?

**Comment by JFDaly** [comment]
Yeah. This is an issue with Workers and Settlers as well as Great People. Workers and Settlers have Capture set as UNITCLASS_WORKER by default, Great People have nil. Workers and Settlers should both become Workers when captured, and Great People should just disappear entirely. But this is what happens instead: 

![Alt Text](https://i.imgur.com/TVEf3t5.jpg)

**Comment by LoneGazebo** [comment]
Oh, I see, it looks like my else if fix didn't make it into production during a manual merge - I probably thought it was an accident and refused the change 🗡

**Comment by LoneGazebo** [comment]
Uploading a debug DLL now...

**Comment by LoneGazebo** [comment]
[CvGameCore_Expansion2.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/2761511/CvGameCore_Expansion2.zip)

**Comment by LoneGazebo** [comment]
Drop that in your CP and see if it now works correctly. It should.

**Comment by LoneGazebo** [comment]
wahoo
